.github/workflows/forum.yml

```diff
@@ -0,0 +1,60 @@
+name: Move to Typst Forum
+on:
+  issues:
+    types:
+      - labeled
+jobs:
+  add-comment:
+    if: github.event.label.name == 'to-forum'
+    runs-on: ubuntu-latest
+    permissions:
+      issues: write
+    steps:
+      - name: Call Discourse API to create forum post
+        env:
+          ISSUE_BODY: ${{ github.event.issue.body }}
+          ISSUE_TITLE: ${{ github.event.issue.title }}
+        run: |
+          RAW="This topic was moved from GitHub issue ${{ github.event.issue.number }}. Please continue the conversation here.
+
+          Here is the original issue:
+          [quote=\"${{ github.event.issue.user.login }}\"]
+          $ISSUE_BODY
+          [/quote]
+
+          Browse the previous discussion at ${{ github.event.issue.html_url }}"
+          TITLE_JSON=$(jq -n --arg title "[#${{ github.event.issue.number }}] $ISSUE_TITLE" '$title')
```

Comment:
Out of abundance of caution, I'd probably move at least these event values (`user.login`, `html_url` and `number`) to environment variables as well, rather than interpolating them in the shell script, since these are potentially user-controlled values (maybe the issue number not so much, but it can receive the same treatment for the sake of consistency). This is important as any kind of shell injection here could leak the Discourse token down below (and any other secrets exposed to this particular script).